File: Entidades/Producto.vb
'Continene: Id_Producto,Descripcion, PrecioUnitario,Ganancia, Existencias, Impuesto, Minimo,
' Maximo, Activo, Consignacion, MaxVendido
Public Class Producto

    'Se declara un constructor para inicializar las lista de los proveedores y los productos relacionados.
    Public Sub New()
        ListaProveedores = New List(Of ProductoPorProveedor)
        ListaRelacionados = New List(Of ProductoRelacionado)
    End Sub

    'Un identificador único para cada producto.
    Private _varIdProducto As Integer
    Public Property Id_Producto() As Integer
        Get
            Return _varIdProducto
        End Get
        Set(ByVal value As Integer)
            _varIdProducto = value
        End Set
    End Property

    'La descripción que muestra el tipo, marca y otras características del producto.
    Private _varDescripcion As String
    Public Property Descripcion() As String
        Get
            Return _varDescripcion
        End Get
        Set(ByVal value As String)
            _varDescripcion = value
        End Set
    End Property

    'El precio por cada unidad del producto.
    Private _varPrecioUnitario As Double
    Public Property PrecioUnitario() As Double
        Get
            Return _varPrecioUnitario
        End Get
        Set(ByVal value As Double)
            _varPrecioUnitario = value
        End Set
    End Property

    'La ganancia que se tendrá sobre cada unidad del producto.
    Private _varGanancia As Double
    Public Property Ganancia() As Double
        Get
            Return _varGanancia
        End Get
        Set(ByVal value As Double)
            _varGanancia = value
        End Set
    End Property

    'El número de unidades disponibles del producto.
    Private _varExistencias As Double
    Public Property Existencias() As Double
        Get
            Return _varExistencias
        End Get
        Set(ByVal value As Double)
            _varExistencias = value
        End Set
    End Property

    'Si se tiene que aplicarle o no impuesto.
    Private _varImpuesto As Boolean
    Public Property Impuesto() As Boolean
        Get
            Return _varImpuesto
        End Get
        Set(ByVal value As Boolean)
            _varImpuesto = value
        End Set
    End Property

    'El valor mínimo de unidades que se deben de tener en stock, se usará para realizar los pedidos.
    Private _varMinimo As Double
    Public Property Minimo() As Double
        Get
            Return _varMinimo
        End Get
        Set(ByVal value As Double)
            _varMinimo = value
        End Set
    End Property

    'Un valor que ayuda a mantener una adecuada cantidad de unidades, para no sobrepasar desmedidamente la cantidad.
    Private _varMaximo As Double
    Public Property Maximo() As Double
        Get
            Return _varMaximo
        End Get
        Set(ByVal value As Double)
            _varMaximo = value
        End Set
    End Property

    'Si el producto esta activo, o ya no se usa.
    Private _varActivo As Boolean
    Public Property Activo() As Boolean
        Get
            Return _varActivo
        End Get
        Set(ByVal value As Boolean)
            _varActivo = value
        End Set
    End Property

    'Si el producto es o no a consignación.
    Private _varConsignacion As Boolean
    Public Property Consignacion() As Boolean
        Get
            Return _varConsignacion
        End Get
        Set(ByVal value As Boolean)
            _varConsignacion = value
        End Set
    End Property

    'El número máximo de unidades vendidas en una sola venta.
    Private _varMaxVendido As Double
    Public Property MaxVendido() As Double
        Get
            Return _varMaxVendido
        End Get
        Set(ByVal value As Double)
            _varMaxVendido = value
        End Set
    End Property

    'Una lista donde se guardarán todos los proveedores de este producto.
    Private _varProveedores As List(Of ProductoPorProveedor)
    Public Property ListaProveedores() As List(Of ProductoPorProveedor)
        Get
            Return _varProveedores
        End Get
        Set(ByVal value As List(Of ProductoPorProveedor))
            _varProveedores = value
        End Set
    End Property

    'Una lista donde se guardan los productos relacionados.
    Private _varRelacionados As List(Of ProductoRelacionado)
    Public Property ListaRelacionados() As List(Of ProductoRelacionado)
        Get
            Return _varRelacionados
        End Get
        Set(ByVal value As List(Of ProductoRelacionado))
            _varRelacionados = value
        End Set
    End Property

    'Obtiene todos los datos que tiene la tabla bodega
    Private _varBodega As Bodega
    Public Property Bodega() As Bodega
        Get
            Return _varBodega
        End Get
        Set(ByVal value As Bodega)
            _varBodega = value
        End Set
    End Property

End Class

File: Entities/Producto.cs
namespace Entidades
{
    using System;
    using System.Collections.Generic;

    public class Producto
    {
        private int _varIdProducto;
        private string _varDescripcion;
        private double _varPrecioUnitario;
        private double _varGanancia;
        private double _varExistencias;
        private bool _varImpuesto;
        private double _varMinimo;
        private double _varMaximo;
        private bool _varActivo;
        private bool _varConsignacion;
        private double _varMaxVendido;
        private List<ProductoPorProveedor> _varProveedores;
        private List<ProductoRelacionado> _varRelacionados;
        private Entidades.Bodega _varBodega;
        private bool _varPrincipal;

        public Producto()
        {
            this.ListaProveedores = new List<ProductoPorProveedor>();
            this.ListaRelacionados = new List<ProductoRelacionado>();
        }

        public int Id_Producto
        {
            get => 
                this._varIdProducto;
            set => 
                this._varIdProducto = value;
        }

        public string Descripcion
        {
            get => 
                this._varDescripcion;
            set => 
                this._varDescripcion = value;
        }

        public double PrecioUnitario
        {
            get => 
                this._varPrecioUnitario;
            set => 
                this._varPrecioUnitario = value;
        }

        public double Ganancia
        {
            get => 
                this._varGanancia;
            set => 
                this._varGanancia = value;
        }

        public double Existencias
        {
            get => 
                this._varExistencias;
            set => 
                this._varExistencias = value;
        }

        public bool Impuesto
        {
            get => 
                this._varImpuesto;
            set => 
                this._varImpuesto = value;
        }

        public double Minimo
        {
            get => 
                this._varMinimo;
            set => 
                this._varMinimo = value;
        }

        public double Maximo
        {
            get => 
                this._varMaximo;
            set => 
                this._varMaximo = value;
        }

        public bool Activo
        {
            get => 
                this._varActivo;
            set => 
                this._varActivo = value;
        }

        public bool Consignacion
        {
            get => 
                this._varConsignacion;
            set => 
                this._varConsignacion = value;
        }

        public double MaxVendido
        {
            get => 
                this._varMaxVendido;
            set => 
                this._varMaxVendido = value;
        }

        public List<ProductoPorProveedor> ListaProveedores
        {
            get => 
                this._varProveedores;
            set => 
                this._varProveedores = value;
        }

        public List<ProductoRelacionado> ListaRelacionados
        {
            get => 
                this._varRelacionados;
            set => 
                this._varRelacionados = value;
        }

        public Entidades.Bodega Bodega
        {
            get => 
                this._varBodega;
            set => 
                this._varBodega = value;
        }

        public bool Principal
        {
            get => 
                this._varPrincipal;
            set => 
                this._varPrincipal = value;
        }
    }
}

